### ISSUE TITLE
Container loses access to PIA (port-forwarding) intermittently and repeatedly: BindPort error / there has been a fatal error

### ISSUE THREAD

**Issue by superkrups20056** [haugene/docker-transmission-openvpn#1445]
My container loses connection randomly.
I am running the latest version of docker and am on the :latest build
It searches for a new port after is loses connection and sometimes it can't find one leading to a fatal error.
I'm running PIA with port forwarding enabled. I use Synology.

The pertinent log is attached. Notice what happens after half an hour after I start my container.

<img width="1422" alt="Screen Shot 2020-10-31 at 9 48 22 PM" src="https://user-images.githubusercontent.com/12375234/97793508-45e2db80-1bc3-11eb-9b53-70140e640ed7.png">
  
[Transmission  copy.txt](https://github.com/haugene/docker-transmission-openvpn/files/5470084/Transmission.copy.txt)

**Comment by JB09** [comment]
Same issue, usually starts happening for me after the container has been running for over 24 hours.

`the has been a fatal_error
the port has been bound to 43540  Mon Nov  2 16:49:31 EST 2020
curl: (28) Connection timed out after 15001 milliseconds
Mon Nov  2 17:10:23 EST 2020: bindPort error `

**Comment by dalloro** [comment]
Same issue.

**Comment by zoidd** [comment]
same issue as well after 24 hours, restating the container temporarily fixes it.

**Comment by ozhound** [comment]
I too have been having this issue for the last couple of days. The container doesnt crash it just starts reporting cant connect to trackers. (im using synology docker)
![image](https://user-images.githubusercontent.com/3649237/98257425-a3f43200-1fba-11eb-8f19-ce2342a6e6e3.png)

That just repeats over and over till i restart the container

I tried using the TCP package from PIA but i cant get it to stick on the container. i delete the ovpn files and copy the tcp ones over and they just get replaced when the container restarts. cant figure out how to force them to stick)

its pretty frustrating.

**Comment by tableofranks** [comment]
Latest docker on arm64 with PIA. Similar issue, however this is on a brand new container and it happens right away.
This occurs with all of the US configs I have tried (US New York, US New Jersey, US Chicago, US Florida, US Atlanta), HOWEVER, the container is able to connect when I use a European config (UK London, France, Spain), so this maybe this is a PIA issue. Compare the logs between US Chicago and France.
[France.txt](https://github.com/haugene/docker-transmission-openvpn/files/5498519/France.txt)
[US Chicago.txt](https://github.com/haugene/docker-transmission-openvpn/files/5498520/US.Chicago.txt)

**Comment by tableofranks** [comment]
It's probably my ISP (WoW) blocking the PIA nextgen US servers..

**Comment by ozhound** [comment]
I cant connect fine to Sweden, Romania and Switzerland. (I've not tried any stateside end points)
it just looses connectivity anywhere between 1-60 minutes.

**Comment by steve-spiteri** [comment]
I face a similar issue using PIA and having the container running for an extended period of time.

You can see in openvpn/pia/update-port.sh the following function:

bind_port () {
  pf_bind=$(curl --insecure --get --silent --show-error \
      --retry $curl_retry --retry-delay $curl_retry_delay --max-time $curl_max_time \
      --data-urlencode "payload=$pf_payload" \
      --data-urlencode "signature=$pf_getsignature" \
      $verify \
      "https://$pf_host:19999/bindPort")
  if [ "$(echo $pf_bind | jq -r .status)" = "OK" ]; then
    echo "the port has been bound to $pf_port  $(date)"		
  else  
    echo "$(date): bindPort error"
    **echo $pf_bind**
    echo "the has been a fatal_error"
  fi
}

Judging from the blank line in the log you can see that the pf_bind variable appears empty. I am still trying to narrow down what is causing the error in the pf_bind variable assignment.

**Comment by ozhound** [comment]
I checked if there where 2 TUN interfaces and there is not. so that rules that out

**Comment by haugene** [comment]
Hey. We've had some stability issues, and PIA has had some too. I would appreciate it if you could test with port-forwarding and health checks disabled. If that runs and is stable then you can try to activate them one by one and see when instability arises. Disable port forwarding by adding the environment variable:
```
DISABLE_PORT_UPDATER=yes
```
To disable health checks add `--no-healthcheck` to your [Docker run](https://docs.docker.com/engine/reference/run/#healthcheck) command or [in docker-compose](https://docs.docker.com/compose/compose-file/#healthcheck):
```
healthcheck:
  disable: true
```

As for the `RESOLVE: Cannot resolve host address` error, that's DNS issue. Try to either specify the --dns flags as described in the documentation. Or add:
```
OVERRIDE_DNS_1=8.8.8.8
OVERRIDE_DNS_2=8.8.4.4
```

**Comment by reprz** [comment]
I´m also having this fatal error issue at the moment. I´m testing now with "--no-healthcheck" enabled. Let´s see what happens.

**Comment by reprz** [comment]
So far I´ve been running this with "--no-healthcheck" enabled for 2 hours without errors. As with healthcheck ON the fatal error could happen even 10 minutes after starting container.

**Comment by haugene** [comment]
@reprz Thanks for letting us know. A PR for a health check fix is in the making #1459

**Comment by superkrups20056** [comment]
Hi guys, I appreciate that my issue received so much traction. I resolved this issue by reverting back to 2.14. 3.0 is a hot mess right now! :)

**Comment by haugene** [comment]
Haha, don't blame you :sweat_smile: However I do believe that the last version is better now :crossed_fingers:
Give it another try, or wait until 2.14 stops working I guess. Close this for now anyhow?
